### ISSUE TITLE
client_email is undefined error after using Workload Identity Provider auth

### ISSUE THREAD

**Issue by pbacon-blaber** [google-github-actions/deploy-cloud-functions#198]
### TL;DR
After following steps in README to deploy using workload identity provider auth, I am getting an error on the deploy step stating "The incoming JSON object does not contain a client_email field"

**Expected behavior**
Deploy would work and not error complaining about JSON auth object.

**Observed behavior**
Got the error described above.

Output:
``` bash
Run google-github-actions/deploy-cloud-functions@v0.6.0

# Prints environment info

Setting project Id from $GCLOUD_PROJECT
Error: The incoming JSON object does not contain a client_email field
```

### Reproduction

**Action YAML**

```yaml
jobs:
  deploy:
    name: Deploy
    runs-on: ubuntu-latest

    # Add "id-token" with the intended permissions.
    permissions:
      contents: "read"
      id-token: "write"

    steps:
      - uses: actions/checkout@v2
      - id: auth
        name: "Authenticate to Google Cloud"
        uses: "google-github-actions/auth@v0.4.1"
        with:
          workload_identity_provider: "projects/{projectId}/locations/global/workloadIdentityPools/{my-pool}/providers/{my-provider}"
          service_account: "{service-account-email}"
      - id: deploy
        uses: google-github-actions/deploy-cloud-functions@v0.6.0
        with:
          name: ${{ env.CLOUD_FN_NAME }}
          runtime: ${{ env.CLOUD_FN_RUNTIME }}
```

**Additional information**
To setup the workload identity provider used in the `google-github-actions/auth@v0.4.1` step, I followed the instructions listed here exactly: https://github.com/google-github-actions/auth#setting-up-workload-identity-federation

**Comment by bharathkkb** [comment]
Hi @pbacon-blaber 
Could you check using `google-github-actions/deploy-cloud-functions@main`. We haven't cut a release yet.

**Comment by pbacon-blaber** [comment]
@bharathkkb that appears to be working for me, thanks so much!

Will this be out in a `0.7.0` release soon?

**Comment by sethvargo** [comment]
Yea, we'll cut a release shortly!
